### PR TITLE
Re-add SchnorrNonce.freshNonce(), for DLC doc. Add mdoc:to-string to …

### PR DESCRIPTION
**Pull request by Christewart**
…dlc doc to make sure the doc is compiled so we have failures on CI if the API changes

### CODE REVIEW

File: core/src/main/scala/org/bitcoins/core/crypto/SchnorrNonce.scala [hunk 1]
```diff
@@ -88,4 +88,9 @@ object SchnorrNonce extends Factory[SchnorrNonce] {
   def apply(xCoor: FieldElement): SchnorrNonce = {
     SchnorrNonce(xCoor.bytes)
   }
+
+  def freshNonce(): SchnorrNonce = {
+    val xcordBytes = ECPrivateKey.freshPrivateKey.publicKey.bytes.tail
+    fromBytes(xcordBytes)
```
Comment (nkohen) [inline core/src/main/scala/org/bitcoins/core/crypto/SchnorrNonce.scala]: nit: `ECPrivateKey.freshPrivateKey.schnorrNonce`

File: docs/wallet/dlc.md [hunk 2]
```diff
@@ -63,8 +64,8 @@ val loseHash = CryptoUtil.sha256(ByteVector("LOSE".getBytes)).flip
 
 (pubKey.bytes ++ rValue.bytes).toHex
 (winHash.bytes ++ Satoshis(100000).bytes ++ loseHash.bytes ++ Satoshis.zero.bytes).toHex
-Schnorr.signWithNonce(winHash.bytes, privKey, nonce).hex
-Schnorr.signWithNonce(loseHash.bytes, privKey, nonce).hex
+NativeSecp256k1.schnorrSignWithNonce(winHash.bytes.toArray, privKey.bytes.toArray, nonce.bytes.toArray)
+NativeSecp256k1.schnorrSignWithNonce(loseHash.bytes.toArray, privKey.bytes.toArray, nonce.bytes.toArray)
```
Comment (nkohen) [inline docs/wallet/dlc.md]: nit: `privKey.schnorrSignWithNonce(_Hash.bytes, nonce)`

File: docs/wallet/dlc.md [hunk 3]
```diff
@@ -48,7 +48,8 @@ f8758d7f03a65b67b90f62301a3554849bde6d00d50e965eb123398de9fd6ea7af05f01f1ca852cf
 
 Note: if you wish to setup your own oracle for testing, you can do so by pasting the following into the `sbt core/console`:
 
-```scala
+```scala mdoc:to-string
+import org.bitcoin._
```
Comment (nkohen) [inline docs/wallet/dlc.md]: This won't be needed given change below

File: docs/wallet/dlc.md (hunk 2 above)
Comment (Christewart) [reply docs/wallet/dlc.md]: So I don't think this works as these methods take a `ECPRivateKey` and not a `SchnorrNonce` 

```scala
  def schnorrSignWithNonce(
      dataToSign: ByteVector,
      nonce: ECPrivateKey): SchnorrDigitalSignature
```

**Pull request merged**